src/common/vfs.cpp

```diff
@@ -69,7 +69,7 @@ Result<bool, QString> Vfs::checkAvailability(const QString &path)
 #ifdef Q_OS_WIN
     if (mode == Mode::WindowsCfApi) {
         const auto info = QFileInfo(path);
-        if (QDir(info.canonicalPath()).isRoot()) {
+        if (QDir(info.canonicalFilePath()).isRoot()) {
```

Comment:
What about 
```bool QFileInfo::isRoot() const```?

Reply:
We need to call canonicalFilePath as explained in https://doc.qt.io/qt-5/qdir.html#isRoot 
Else C:\\..\\ is not reported as a root drive.
I don't think it will make much difference whether we create a temporary QDir or a temporary QFileInfo

Reply:
Can't you just 
```
if (info.isRoot()) {
...
}```
without constructing any temporary object?

Reply:
bool QFileInfo::isRoot() const

Returns true if the object points to a directory or **to a symbolic link to a directory**, and that directory is the root directory; otherwise returns false.

So I assume QFileInfo::isRoot() would cover the case regarding canonicalFilePath

Reply:
```
int main(int argc, char *argv[])
{
    QApplication app(argc, argv);

    auto foo = [](const QString &s){
        QFileInfo info(s);
        qDebug() << s << info.canonicalFilePath();
        qDebug() << info.isRoot();
        qDebug() << QDir(info.canonicalFilePath()).isRoot();
    };
    foo("C:\\");
    foo("C:\\..\\");
    foo("C:");
    return app.exec();
}
```


```
"C:\\" "C:/"
true
true
"C:\\..\\" "C:/"
false
true
"C:" "C:/Users/hanna/Documents/build-untitled-ownCloud-Debug"
false
false
```

Reply:
Thanks for the test example. Then I think this seems to be a Qt bug. At least based on documentation I would expect a different behavior.

Reply:
FYI https://bugreports.qt.io/browse/QTBUG-94582